rusk-wallet/src/main.rs

```diff
@@ -0,0 +1,348 @@
+        /// Key index from which to stake Dusk
+        #[clap(short, long)]
+        key: u64,
+
+        /// Staking key to sign this stake
+        #[clap(short, long)]
+        stake_key: u64,
+
+        /// Amount of Dusk to stake
+        #[clap(short, long)]
+        amt: u64,
+
+        /// Max amt of gas for this transaction
+        #[clap(short = 'l', long)]
+        gas_limit: u64,
+
+        /// Max price you're willing to pay for gas used
+        #[clap(short = 'p', long)]
+        gas_price: Option<u64>,
+    },
+
+    /// Extend stake for a particular key
+    ExtendStake {
+        /// Key index from which your Dusk was staked
+        #[clap(short, long)]
+        key: u64,
+
+        /// Staking key index used for this stake
+        #[clap(short, long)]
+        stake_key: u64,
+
+        /// Max amt of gas for this transaction
+        #[clap(short = 'l', long)]
+        gas_limit: u64,
+
+        /// Max price you're willing to pay for gas used
+        #[clap(short = 'p', long)]
+        gas_price: Option<u64>,
+    },
+
+    /// Withdraw a key's stake
+    WithdrawStake {
+        /// Key index from which your Dusk was staked
+        #[clap(short, long)]
+        key: u64,
+
+        /// Staking key index used for this stake
+        #[clap(short, long)]
+        stake_key: u64,
+
+        /// Max amt of gas for this transaction
+        #[clap(short = 'l', long)]
+        gas_limit: u64,
+
+        /// Max price you're willing to pay for gas used
+        #[clap(short = 'p', long)]
+        gas_price: Option<u64>,
+    },
+
+    None
+}
+
+impl CliCommand {
+    fn uses_wallet(&self) -> bool {
+        match *self {
+            Self::Create | Self::Restore | Self::None => false,
+            _ => true,
+        }
+    }
+}
+
+impl WalletCfg {
+
+    /// Default data directory is in user's home dir
+    fn default_data_dir() -> String {
+        let home = dirs::home_dir().expect("OS not supported");
+        let path = Path::new(home.as_os_str()).join(DATA_DIR);
+        String::from(path.to_str().unwrap())
+    }
+
+    /// Default wallet name is essentially current username
+    fn default_wallet_name() -> String {
+        // get default user as default wallet name (remove whitespace)
+        let mut user: String = whoami::username();
+        user.retain(|c|!c.is_whitespace());
+        user.push_str(".dat");
+        user
+    }
+
+    /// Checks consistency of loaded configuration
+    fn sanity_check(&self) -> Result<(), CliError> {
+        // TODO!
+        Ok(())
+    }
+
+}
+
+#[tokio::main]
+async fn main() -> Result<(), CliError> {
+
+    // parse cli arguments
+    let cfg: WalletCfg = WalletCfg::parse();
+
+    cfg.sanity_check()?;
+
+    // some commands don't use an existing wallet
+    // get those out of the way first
+    let cmd = cfg.command.unwrap_or_else(|| CliCommand::None);
+    if !cmd.uses_wallet() {
+        use CliCommand::*;
+        return match cmd {
+            Create => create(),
```

Comment:
Need to pass arguments to `create()`, at least all path-related stuff in case user overrides